addons/mail/models/discuss/discuss_channel.py

```diff
@@ -1112,6 +1112,9 @@ def channel_fetched(self):
                 return
             last_message_id = channel.message_ids.ids[0] # zero is the index of the last message
             member = self.env['discuss.channel.member'].search([('channel_id', '=', channel.id), ('partner_id', '=', self.env.user.partner_id.id)], limit=1)
+            if not member:
+                # member not a part of the channel
+                return
```

Comment:
All of the `return` in this method should be `continue` instead as they are in a loop.
I think we only ever call it on a single record that's why the issue is not really visible.

Reply:
Logically they would perform the same as the `self` recordset is always of length 1 in this case.

Reply:
Not a good reason to write it incorrectly, if you could take the opportunity to make them all `continue` it would be great.

Reply:
Added the change!